apps/dashboard/lib/trpc/routers/key/updateRemaining.ts

```diff
@@ -57,6 +58,7 @@ export const updateKeyRemaining = rateLimitedProcedure(ratelimit.update)
           input.refill?.interval === "none" || input.refill?.interval === undefined
             ? null
             : input.refill?.interval,
+        refillDay: input.refill?.interval === "monthly" ? input.refill.refillDay : undefined,
```

Comment:
setting `refillDay: undefined` will not overwrite the value in the db, is that what we want here?
it looks like what we actually want is to set it to null, no? I might be wrong here